### ISSUE TITLE
Usage with ProBuilder Meshes

### ISSUE THREAD

**Issue by Monitor9** [naelstrof/SkinnedMeshDecals#15]
Hello, I use SkinnedMeshDecals on simple meshes with MeshFilter and MeshRenderer, as well as on meshes created with ProBuilder. I was able to generate UV2s lightmaps for these meshes, but even with maximum Texel Per Meter, their quality is terribly low. 
**Example:**
![image](https://user-images.githubusercontent.com/132024693/235016814-cba2299c-569d-41d7-ae39-991e565850bd.png)
**ProBuilder UV2 creation:**
![image](https://user-images.githubusercontent.com/132024693/235017029-12575493-193b-4467-b563-c7f7c34ccc70.png)
**Shader:**
![image](https://user-images.githubusercontent.com/132024693/235018262-c64db59f-3eef-4fb2-a0fa-3f156559a085.png)
Is this due to the fact that the meshes can be very large in size, or did I set something up incorrectly?

Please forgive my poor English, as it is generated by the ChatGpt translation system.

**Comment by naelstrof** [comment]
Yeah this is an unfortunate side effect of the strategy, large meshes can't possibly get enough resolution. You have a couple of options though:

- Fall back to regular decals (URP screen space decals for example) when the mesh size is large.

- Use the texture as a "damage mask" rather than raw texture printing. Allowing you to use the _decalcolomap texture as a mask to sample some high-resolution damage from another repeated texture.

- Break up the level into smaller chunks.

I hope that helps!

**Comment by Monitor9** [comment]
Thank you for your response. I'd prefer not to use the first option, as the standard Unity Decal Projectors on the LTS version don't support layer filtering, and don't filter meshes at all (they project onto everything, and I have a lot of moving props in the game, so they'll be affected).
![test](https://user-images.githubusercontent.com/132024693/235189428-1f828b61-23b6-4140-90d0-fd92557652bc.gif)
Can you provide more details on the second option? Perhaps an article or video, as I'm not quite sure what you're referring to.

**Comment by naelstrof** [comment]
I don't have articles or videos, but an example of what it might look like:
![image](https://user-images.githubusercontent.com/1131571/235583195-c4ce671f-8a3f-4955-942f-2fcdc240973f.png)

In essence you'd project a pure red dot as a decal onto the mesh, then use the red channel of the _DecalColorMap as a mask to reveal damaged parts.

Depending on the kind of damage, you can do different things, Red could be gradual damage that removes skin/muscle until it reveals bones. Green could be water wetness (for bodies of water), and Blue could be muddiness/dirtiness.

Just use them as masks to sample underlying triplanar maps, or cool glowy effects, whatever.

Then you'd have no need for traditional decals, because you'd simply author decals as how hurty/wet/muddy they are.
